### ISSUE TITLE
Output Typescript Code

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite-api#9]
The code for generating the API could still be Javascript, since it only affects us. 

However, a lot of people may want to use the output JS API in Typescript projects (which offer superior autocompletion than vanilla JS).  Thus, it might be worth making sure that our output code is in TS.  

cc: @domoritz

**Comment by RandomFractals** [comment]
yeah, I'd be curious to see TS setup for this lib api too, not just for JS gen. I prefer to code in TS. 

what's required to make this lib work in TS?

**Comment by RandomFractals** [comment]
@kanitw @jheer & @domoritz do you guys still plan to work on the companion https://github.com/vega/vega-lite-to-api repo?

I think it would be great to be able to have a library that generates this vega-lite api code from Vega specs.

I'd love to add that feature to the Vega Viewer 📈 when you are ready for it ...

https://github.com/RandomFractals/vscode-vega-viewer

**Comment by kanitw** [comment]
@RandomFractals we will get back to it that some points, but currently doesn't have time to finish it.

**Comment by RandomFractals** [comment]
:( ok

**Comment by RandomFractals** [comment]
any ETA on this? if you don't think you'll get to it this year, I'd rather close that issue in my repo.

**Comment by domoritz** [comment]
We’re more than happy to accept a PR for this feature. I don’t think we have a pressing need for this from our side before the end of the year.

**Comment by RandomFractals** [comment]
I'd have to play a bit with this api library to see what needs to be done for the TS code gen. I doubt I'll get to it this year also, but let's keep this open. I think it would be a great feature to have eventually.
